packages/uicore/src/components/DropDown/DropDown.tsx

```diff
@@ -90,46 +99,52 @@ export const DropDown: React.FC<DropDownProps> = props => {
     addClearBtn = false,
     getCustomLabel,
     disabled,
+    isLazyItemsQuery,
     ...rest
   } = props
 
-  const [dropDownItems, setDropDownItems] = React.useState<SelectOption[]>([])
-  const [loading, setLoading] = React.useState<boolean>(false)
-  const [internalQuery, setInternalQuery] = React.useState<string>('')
-  const [activeItem, setActiveItem] = React.useState<SelectOption | null>(null)
+  const [dropDownItems, setDropDownItems] = useState<SelectOption[]>([])
+  const [loading, setLoading] = useState<boolean>(false)
+  const [internalQuery, setInternalQuery] = useState<string>('')
+  const [isDropdownOpenedOnce, setDropdownOpenedOnce] = React.useState(false)
 
-  React.useEffect(() => {
+  const activeItem = useMemo(
+    () =>
+      dropDownItems.find(item => item.value === value?.toString()) || {
+        label: '',
+        value: ''
+      },
+    [value, dropDownItems]
+  )
+
+  useEffect(() => {
     if (Array.isArray(items)) {
-      setDropDownItems([...items])
+      setDropDownItems(items)
     } else if (typeof items === 'function' && !loading) {
-      // Do not enter this block if already loading
       setLoading(true)
-      const promise = items()
-
-      if (typeof promise.then === 'function') {
-        promise.then(results => {
-          setDropDownItems(results)
-          setLoading(false)
+      Promise.resolve(items())
+        .then(itemsResponse => {
+          setDropDownItems(itemsResponse)
         })
-      }
-      if (typeof promise.catch === 'function') {
-        promise.catch(errorResults => {
-          setDropDownItems(errorResults)
+        .finally(() => {
           setLoading(false)
         })
-      } else {
-        setLoading(false)
-      }
     }
-  }, [query, JSON.stringify(items)])
+  }, [JSON.stringify(items)])
 
-  React.useEffect(() => {
-    const newActiveItem = dropDownItems.find(item => item.value === value?.toString()) || {
-      label: '',
-      value: ''
+  useEffect(() => {
+    // For lazy loaded queries call the items function once, only if dropdown is opened
+    if (isLazyItemsQuery && isDropdownOpenedOnce && dropDownItems.length === 0 && typeof items === 'function') {
+      setLoading(true)
+      Promise.resolve(items())
+        .then(itemsResponse => {
+          setDropDownItems(itemsResponse)
+        })
+        .finally(() => {
+          setLoading(false)
+        })
```

Comment:
Nit: this block can be moved to a function, as it is also used in the above effect

Reply:
Existing block. Not used anywhere else?